wrapanapi/systems/msazure.py

```diff
@@ -1111,3 +1143,15 @@ def is_stack_empty(self, stack_name, resource_group):
 
     def info(self):
         pass
+
+    def cleanup_resources(self):
+        """
+        Clean up Unused Disks, Nics and Public IPs
+
+        Any exceptions raised during NIC/PIP/Disc delete attempts are logged only.
+        """
+        self.logger.info("cleanup: Removing all unused NICs/PIPs/Discs in resource group")
+        self.remove_nics_by_search(self.resource_group)
```

Comment:
The functions each factor in `self.resource_group`, if you're going to include it here it should be taken as an argument and passed through. Otherwise just leave it off and let the functions add in the resource group.

Reply:
Removed argument to the functions as the functions themselves will use the constructor initiated resource group.